### PR TITLE
librespot 0.2.0 (new formula)

### PR DESCRIPTION
**Pull request by cprecioso**
- [x] Have you followed the [guidelines for contributing](https://github.com/Homebrew/homebrew-core/blob/HEAD/CONTRIBUTING.md)?
- [x] Have you ensured that your commits follow the [commit style guide](https://docs.brew.sh/Formula-Cookbook#commit)?
- [x] Have you checked that there aren't other open [pull requests](https://github.com/Homebrew/homebrew-core/pulls) for the same formula update/change?
- [x] Have you built your formula locally with `brew install --build-from-source <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Is your test running fine `brew test <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Does your build pass `brew audit --strict <formula>` (after doing `brew install --build-from-source <formula>`)? If this is a new formula, does it pass `brew audit --new <formula>`?

-----

The test is lacking, as I haven't found a way of making librespot just try to connect and then exit. It just keeps running until you manually close it, so I don't know how to test it.

### CODE REVIEW

File: Formula/librespot.rb [hunk 1]
```diff
@@ -0,0 +1,20 @@
+class Librespot < Formula
+  desc "Open Source Spotify client library"
+  homepage "https://github.com/librespot-org/librespot"
+  url "https://github.com/librespot-org/librespot.git",
+      tag:      "v0.2.0",
+      revision: "59683d7965480e63c581dd03082ded6a080a1cd3"
```
Comment (carlocab) [inline Formula/librespot.rb:6]: Can we use a tarball instead of a Git clone?

File: Formula/librespot.rb [hunk 2]
```diff
@@ -0,0 +1,20 @@
...
+  license "MIT"
+  head "https://github.com/librespot-org/librespot.git",
+      branch: "dev"
```
Comment (carlocab) [inline Formula/librespot.rb:9]:
```suggestion
  head "https://github.com/librespot-org/librespot.git", branch: "dev"
```

File: Formula/librespot.rb [hunk 3]
```diff
@@ -0,0 +1,20 @@
...
+
+  depends_on "rust" => :build
+
+  def install
+    system "cargo", "install", "--no-default-features", "--features", "rodio-backend,with-dns-sd", *std_cargo_args
+  end
+
+  test do
+    system "#{bin}/librespot", "-V"
```
Comment (carlocab) [inline Formula/librespot.rb:18]: Can you propose a more substantial test than checking the version or calling for help? These tests rarely ever fail even for broken installations.

>We want tests that don’t require any user input and test the basic functionality of the application. For example `foo build-foo input.foo` is a good test and (despite their widespread use) `foo --version` and `foo --help` are bad tests. 

Ref. https://docs.brew.sh/Formula-Cookbook#add-a-test-to-the-formula

Comment (cprecioso) [reply Formula/librespot.rb:18]: Yeah, I know, but as I wrote in the PR description:

> The test is lacking, as I haven't found a way of making librespot just try to connect and then exit. It just keeps running until you manually close it, so I don't know how to test it.

Comment (carlocab) [reply Formula/librespot.rb:16]: Can't we run it for a little while, get it to dump some output we can verify, then quit?

**Comment by cprecioso** [comment]
I added a test, it works fine, but please do check it as I am not fluent in Ruby

**Comment by chenrui333** [comment]
failed to build on Mojave and Catalina
```
error: failed to run custom build command for `coreaudio-sys v0.2.8`
```

**Comment by cprecioso** [comment]
Unfortunately I do not have access to a Mojave or Catalina system to debug that, or sufficient knowledge of Rust to know what would be happening :/

**Comment by dawidd6** [comment]
FTFY

**Review by SMillerDev — APPROVED** [review]
Thanks @cprecioso  ! Without contributions like yours it'd be impossible to keep homebrew going with the high standards that users have come to expect from the project. You can feel good knowing that you've made the world a tiny bit better for homebrew users around the world! 👍 🎉

**Pull request closed without merging**